src/main/java/ui/components/pickers/MilestonePicker.java

```diff
@@ -0,0 +1,44 @@
+package ui.components.pickers;
+
+import backend.resource.TurboIssue;
+import backend.resource.TurboMilestone;
+import javafx.application.Platform;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
+import javafx.util.Pair;
+import ui.UI;
+import undo.actions.ChangeMilestoneAction;
+import util.events.ShowMilestonePickerEventHandler;
+
+import java.util.List;
+import java.util.Optional;
+
+public class MilestonePicker {
+    UI ui;
+    Stage stage;
+
+    public MilestonePicker(UI ui, Stage mainStage) {
+        this.ui = ui;
+        this.stage = mainStage;
+        ui.registerEvent((ShowMilestonePickerEventHandler) e -> Platform.runLater(() -> showMilestonePicker(e.issue)));
+    }
+
+    private void showMilestonePicker(TurboIssue issue) {
+        List<TurboMilestone> milestoneList = ui.logic.getRepo(issue.getRepoId()).getMilestones();
```

Comment:
Types are metadata/implementation details, and can be left out of variable names.